### PR TITLE
Add initial user event tests

### PR DESCRIPTION
**Pull request by ozzysChiefDataScientist**
I added an initial test to check if the message "Loading..." appears after someone hits "Submit" on the search form.

### CODE REVIEW

File: client/src/__tests__/search.test.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import * as React from 'react'
+import {render, screen, waitFor} from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import "mutationobserver-shim"
+import Search from '../components/Search'
+
+jest.setTimeout(30000);
```
Comment (laredotornado) [inline client/src/__tests__/search.test.js:7]: Why do we need this?

File: client/src/__tests__/search.test.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+test('test if "loading" message appears after submitting a search', async () => {
+
+  render(<Search />)
+
+  screen.getByRole('textbox', { name: 'Name' });
+  userEvent.type(screen.getByRole('textbox', { name: 'Name' }), '1335 ASTOR');
+  userEvent.click(screen.getByRole('button', {name: /Submit/i}));
+
+  await waitFor(() => screen.findByText(/Loading/i), {timeout:5000});
```
Comment (laredotornado) [inline client/src/__tests__/search.test.js:17]: What is the timeout for here?  Shouldn't the loading message show up immediately upon form submission and then disappear when results are returned?

**Review by laredotornado — COMMENTED** [review]
Thanks for taking charge on this.  I think one test to add to this suite would be to mock the results coming back from the AJAX call to the search endpoint.  The test would verify that this call was invoked and that results were rendered properly.

File: client/src/__tests__/search.test.js (hunk 1 above)
Comment (ozzysChiefDataScientist) [reply client/src/__tests__/search.test.js:7]: Hi @laredotornado  - Are you asking about the mutationobserver-shim? I added it for a few reasons:
* When I run the test without it, I get the error " TypeError: MutationObserver is not a constructor" on the line "await waitFor(() => screen.findByText(/Loading/i))."
* I found mutationobserver-shim as a solution on this StackOverflow post: https://stackoverflow.com/questions/64188000/mutation-observer-is-not-a-constructor-jest.
* Some other StackOverflow posts have suggested adding a devDependency on "jest-environment-jsdom-sixteen", but I didn't try that once I found that mutationobserver-shim worked. I can try jest-environment-jsdom-sixteen if you prefer. (StackOverflow link: https://stackoverflow.com/questions/61036156/react-typescript-testing-typeerror-mutationobserver-is-not-a-constructor)

File: client/src/__tests__/search.test.js (hunk 2 above)
Comment (ozzysChiefDataScientist) [reply client/src/__tests__/search.test.js]: I removed the timeout.

**Comment by ozzysChiefDataScientist** [comment]
@laredotornado -- I agree we should have a test that checks if results are rendered. I tried working on this weekend but haven't been able to get the test to pass. 

Here's my current working draft of the test: https://github.com/chicommons/maps/pull/140/commits/0059f011d27595f94b376fe47c6658efb6dc4c04.

Here's a preview of the error message I've been getting. It looks like the test is never seeing the word "Results" appear, even if I set a very high timeout threshold on the test. I will need to spend more time learning about testing concepts to figure out how to resolve this.

results_test

**Review by laredotornado — APPROVED** [review]

**Pull request merged**